autoendpoint/src/routers/fcm/router.rs

```diff
@@ -106,20 +108,36 @@ impl Router for FcmRouter {
             .router_data
             .as_ref()
             .ok_or(FcmError::NoRegistrationToken)?;
-        let fcm_token = router_data
-            .get("token")
-            .and_then(Value::as_str)
-            .ok_or(FcmError::NoRegistrationToken)?;
-        let app_id = router_data
-            .get("app_id")
-            .and_then(Value::as_str)
-            .ok_or(FcmError::NoAppId)?;
+
+        // Older "GCM" set the router data as "auth", "senderID"
+        // Newer "FCM" set the router data as "token", "app_id"
+        // Try reading as FCM and fall back to GCM.
+        let fcm_token = match router_data.get("token").and_then(Value::as_str) {
+            Some(v) => v,
+            None => router_data
+                .get("auth")
+                .and_then(Value::as_str)
+                .ok_or(FcmError::NoRegistrationToken)?,
+        };
+        let app_id = match router_data.get("app_id").and_then(Value::as_str) {
```

Comment:
Are we sure these are the sole adjustments needed? The GCM vs FCM payload and Auth header look slightly different.